internal/secretref/secret_ref.go

```diff
@@ -28,6 +30,19 @@ func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, gett
 	return false
 }
 
+func ValidateAndSanitizeTLSSecret(tlsCert, tlsKey []byte) ([]byte, []byte) {
+	_, err := tls.X509KeyPair(tlsCert, tlsKey)
+	if err != nil {
+		if noPemDataFoundError(err) {
```

Comment:
I would do the replacement in the case of any error. Relying on the message is not very robust and this replacement should usually  not cause any harm.